tribe-ext-settings-import-export.php

```diff
@@ -253,76 +233,94 @@ function tribe_sie_process_settings_import() {
 				return;
 			}
 
-			$import_file = $_FILES['import_file']['tmp_name'];
-			$import_filename = $_FILES['import_file']['name'];
-
-			if ( empty( $import_file ) ) {
-				wp_die( __( 'Please upload a file to import.', 'tribe-ext-settings-import-export' ) );
+			/**
+			 * Export actions.
+			 * Process a settings export that generates a .json file of the shop settings.
+			 */
+			if ( ! empty ( $_POST['export'] ) ) {
+				$settings = get_option( 'tribe_events_calendar_options' );
+
+				// TEC - widget_tribe-events-list-widget
+				// PRO - widget_tribe-events-adv-list-widget
+				// PRO - widget_tribe-events-countdown-widget
+				// PRO - widget_tribe-mini-calendar
+				// PRO - widget_tribe-events-venue-widget
+				// PRO - widget_tribe-events-venue-widget
+				// CE  - tribe_community_events_options
+				// CE  - Tribe__Events__Community__Schemaschema_version
+
+				ignore_user_abort( true );
+				nocache_headers();
+				header( 'Content-Type: application/json; charset=utf-8' );
+				header( 'Content-Disposition: attachment; filename=tribe-settings-export-' . date( 'm-d-Y' ) . '.json' );
+				header( "Expires: 0" );
+				echo json_encode( $settings );
 			}
 
-			if ( ! empty( $import_filename ) ) {
-				$tmp = explode( '.', $import_filename );
-				$extension = end( $tmp );
-			}
+			/**
+			 * Import actions.
+			 * Process a settings import from a json file.
+			 */
+			if ( ! empty ( $_POST['import'] ) ) {
 
-			if ( ! isset ( $extension ) || $extension != 'json' ) {
-				wp_die( __( 'Please upload a valid .json file.', 'tribe-ext-settings-import-export' ) );
-			}
+				$import_file = $_FILES['import_file']['tmp_name'];
+				$import_filename = $_FILES['import_file']['name'];
```

Comment:
Though we expect `$_FILES['import_file']` to be set and contain the keys you referenced, that may not be the case. I realize you test if the vars are empty just a few lines down, but we could avoid error log noise by doing something like this:

```php
$import_file = Tribe__Utils__Array::get( $_FILES, [ 'import_file', 'tmp_name' ], false );
$import_filename = Tribe__Utils__Array::get( $_FILES, [ 'import_file', 'name' ], false );
```

Tribe Common provides this helper and we know it will be available. The core difference is that, this way, we make sure $_FILES is an array and that it contains `import_file` and that `import_file` is an array that contains `tmp_name` and `name`.

(I wrote the above on the fly, so if you adopt this suggestion please test/validate the call is correct.)

Reply:
Okay, this doesn't want to work for now.
I added `use Tribe__Utils__Array;` at the beginning - thought I would need that.
Now I'm getting `Please upload a file to import`.

Reply:
Solved it.